tests/unit/drivers/rank/aggregate/test_chunk2doc_rank_drivers.py

```diff
@@ -21,13 +21,23 @@ def _get_score(self, match_idx, query_chunk_meta, match_chunk_meta, *args, **kwa
 
 class MockLengthRanker(Chunk2DocRanker):
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.required_keys = {'length'}
+        super().__init__(query_required_keys=['length'], match_required_keys=['length'], *args, **kwargs)
 
     def _get_score(self, match_idx, query_chunk_meta, match_chunk_meta, *args, **kwargs):
         return match_idx[0][self.COL_PARENT_ID], match_chunk_meta[match_idx[0][self.COL_DOC_CHUNK_ID]]['length']
 
 
+class MockPriceDiscountRanker(Chunk2DocRanker):
+    def __init__(self, *args, **kwargs):
+        super().__init__(query_required_keys=('tags__price', 'tags__discount'), match_required_keys=('tags__price', 'tags__discount'), *args, **kwargs)
+
+    def _get_score(self, match_idx, query_chunk_meta, match_chunk_meta, *args, **kwargs):
+        price = match_chunk_meta[match_idx[0][self.COL_DOC_CHUNK_ID]]['tags__price']
+        discount = match_chunk_meta[match_idx[0][self.COL_DOC_CHUNK_ID]]['tags__discount']
+        return match_idx[0][self.COL_PARENT_ID], \
+               price - price * discount
+
+
 class SimpleChunk2DocRankDriver(Chunk2DocRankDriver):
 
```

Comment:
discount = 0.5

Reply:
But here I want to shpw that is passed to the Ranker

Reply:
here the same